python_src/models/simulation.py

```diff
@@ -43,6 +45,42 @@ def collect_vehicle_position(player: models.Player) -> models.Player:
 
         return player
 
+    def get_vehicles_distance(self, debug: bool = False) -> float:
+        v1, v2 = self.players[0].vehicle, self.players[1].vehicle
+        p1, p2 = Point(v1.state['pos'][0], v1.state['pos'][1]), Point(v2.state['pos'][0], v2.state['pos'][1])
+        distance = p1.distance(p2)
+
+        # Debug line
+        if debug is True:
+            print("Distances between vehicles: ", distance)
+
+        return distance
+
+    @staticmethod
+    def trigger_vehicle(player: models.Player, distance_report: float = None, debug: bool = False) -> bool:
+        is_trigger = False
+        # The car stills wait until their current distance <= distance_to_trigger
+        if distance_report is not None and player.distance_to_trigger > distance_report:
+            is_trigger = True
+
+        # Trigger normal vehicles which move in the beginning
+        if distance_report is None and player.distance_to_trigger == -1:
+            is_trigger = True
+
+        # Add vehicle to a scenario
+        if is_trigger:
+            vehicle = player.vehicle
+            road_pf = player.road_pf
+            if len(road_pf.script) > 2:
+                vehicle.ai_set_mode("manual")
+                vehicle.ai_set_script(road_pf.script, cling=False)
```

Comment:
I see, so we are using a trigger based on distance rather than time. Is this because AC3R implements it that way or this decision was motivated by some other means?

 I guess this is ok for now, I wonder how this may scale if we need to move more than 2 vehicles.

Reply:
Because AC3R implements the idle timing by triggering distance, regarding the scaling problem, it would be best to implement a trigger by timing instead of distance.